### PR TITLE
add documentation queue factory [#159909975]

### PR DESCRIPTION
**Pull request by ajzoell**
Updated QueueFactory and IQueueFactory with documentation comments

### CODE REVIEW

File: src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs [hunk 1]
```diff
@@ -5,15 +5,31 @@
...
+        /// <param name="connection">Connection string enabling database connection.</param>
         public QueueFactory(string connection)
         {
             connectionString = connection;
         }
 
+        /// <summary>
+        /// Creates new QueueClient object used to handle asynchronous calls to another Web API.
+        /// </summary>
+        /// <param name="queueName">Name given to the QueueClient object.</param>
```
Comment (Hoagithor) [inline src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs]: "queueName" is the name of the Queue being connected to, not the queue client

File: src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs [hunk 2]
```diff
@@ -5,15 +5,31 @@
 using Microsoft.Azure.ServiceBus;
 namespace ServiceHub.User.Service.ServiceBus
 {
+    /// <summary>
+    /// Class designed to create new QueueClient object. This object is responsible for handling
+    /// asynchronous calls to other Web APIs.
+    /// </summary>
```
Comment (Hoagithor) [inline src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs]: ..new QueueClient objects*. 
QueueClients are responsible for handling messages to the ServiceBus.* 
Again, sorry, I should have wrote something or discussed with you.

File: src/ServiceHub.User.Service/ServiceBus/IQueueFactory.cs [hunk 3]
```diff
@@ -6,6 +6,9 @@
 
 namespace ServiceHub.User.Service.ServiceBus
 {
+    /// <summary>
+    /// Interface created to allow mocking of the QueueFactory class.
+    /// </summary>
     public interface IQueueFactory
     {
         IQueueClient GetQueueClient(string queueName);
```
Comment (Hoagithor) [inline src/ServiceHub.User.Service/ServiceBus/IQueueFactory.cs]: Method needs doc. I think the QueueFactory class doc could go here and you could inherit the doc in the class.

**Comment by ajzoell** [comment]
Changes to the _site folder were autogenerated. No need to check them.

File: src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs [hunk 4]
```diff
@@ -5,15 +5,29 @@
...
+    /// handling messages to the ServiceBus.
+    /// </summary>
     public class QueueFactory : IQueueFactory
     {
         private readonly string connectionString;
 
+        /// <summary>
+        /// Constructor for the QueueFactory class.
+        /// </summary>
+        /// <param name="connection">Connection string enabling database connection.</param>
```
Comment (Hoagithor) [inline src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs]: enabling ServiceBus connection

File: src/ServiceHub.User.Service/ServiceBus/IQueueFactory.cs [hunk 5]
```diff
@@ -6,8 +6,19 @@
...
+    /// Interface created to allow mocking of the QueueFactory class.
+    /// </summary>
     public interface IQueueFactory
     {
+        /// <summary>
+        /// Creates new QueueClient object used to handle asynchronous calls to another Web API.
+        /// </summary>
+        /// <param name="queueName">Name of queue being connected to.</param>
+        /// <returns>
+        /// New QueueClient object with name equal to the value of queueName and
```
Comment (Hoagithor) [inline src/ServiceHub.User.Service/ServiceBus/IQueueFactory.cs]: "with name equal" -> "with QueueName equal"

File: src/ServiceHub.User.Service/ServiceBus/IQueueFactory.cs [hunk 6]
```diff
@@ -6,8 +6,19 @@
 
 namespace ServiceHub.User.Service.ServiceBus
 {
+    /// <summary>
+    /// Interface created to allow mocking of the QueueFactory class.
+    /// </summary>
     public interface IQueueFactory
     {
+        /// <summary>
+        /// Creates new QueueClient object used to handle asynchronous calls to another Web API.
```
Comment (Hoagithor) [inline src/ServiceHub.User.Service/ServiceBus/IQueueFactory.cs]: "to another WebAPI" -> "to the ServiceBus"

File: src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs [hunk 7]
```diff
@@ -5,15 +5,29 @@
...
         {
             connectionString = connection;
         }
 
+        ///<inheritdoc />
+        //Above tag inherits the documentation comments from the implemented interface. When you run the
+        //command "docfx serve _site", the documentation will show the proper summary and parameter
+        //description, but it will not put those comments in the source code. To serve docfx, you must be
+        //in the same folder as the _site folder in you repository, then run said docfx command. You may need
+        //to run the command "docfx" first if the serve command fails.
```
Comment (escalonn) [inline src/ServiceHub.User.Service/ServiceBus/QueueFactory.cs]: do we need docs on docfx itself here? also should probably be space after ///

**Comment by ajzoell** [comment]
DocFX will only track comments beginning with "///". Any comment starting with "//" will not be tracked by DocFX. The "//" was added to help clarify how to check that the documentation is inherited by the QueueFactory class for the GetQueueClient method.

**Pull request merged**